External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -3582,7 +3587,17 @@ private static ComaExpression DisambiguateComa(ScintillaControl Sci, int positio
                         if (":,(=".IndexOf(c) >= 0)
                         {
                             string line = Sci.GetLine(Sci.LineFromPosition(position));
+                            //TODO: Very limited check, the case|default could be in a previous line, or it could be something else in the same line
                             if (Regex.IsMatch(line, @"\b(case|default)\b.*:")) break; // case: code block
+                            if (c == ':' && Sci.ConfigurationLanguage == "haxe")
```

Comment:
We need to replace all the `ASContext.Context.CurrentModel.haXe`, `ASContext.Context.CurrentModel.Version == 3`, `Sci.ConfigurationLanguage == "haxe"` by something less heuristic.